### PR TITLE
Created an class to generate stores based on config and the test class t...

### PR DESCRIPTION
**Pull request by jzeeck**
...hat tests the current implementation, this is ready for the CassandraStore.

### CODE REVIEW

File: src/core/StoreSupplier.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Use this class so create a TsdbStore instance. Given a config file this
+ * class will set up an instance of the configured type of the database store.
+ */
+public class StoreSupplier implements Supplier{
+
+    private static final org.slf4j.Logger LOG = LoggerFactory.getLogger
```
Comment (luuse) [inline src/core/StoreSupplier.java]: Can you clean up the type with an import and rename the constant to `LOGGER` as this name is used everywhere else?

File: src/core/StoreSupplier.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            (StoreSupplier.class);
+
+    /**
+     * The config object to be used when we want to create a
+     * {@link TsdbStore} object.
+     */
+    private final Config config;
```
Comment (luuse) [inline src/core/StoreSupplier.java]: Can you put this instance variable bellow all constants?

File: src/core/StoreSupplier.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+     * @return This method will return a ready to use {@link TsdbStore} object.
+     * No guarantee is made that it will connect properly to the database but
+     * it will be configured according to the config class sent in when this
+     * object was created.
+     */
+    @Override
+    public TsdbStore get() {
+        String adapter_type = config.getString("tsd.storage.adapter");
+        if (Strings.isNullOrEmpty(adapter_type)) {
+            LOG.error("The config could not find the" +
```
Comment (luuse) [inline src/core/StoreSupplier.java]: If the exception you throw bellow is handled correctly (it's logged) then this log message is redundant and provides less information on its own than the exception will, thus I'd rather remove this log statement completely.

File: src/core/StoreSupplier.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        }
+
+
+        if (HBASE_NAME.equals(adapter_type.toLowerCase()))
+            return getHBaseStore();
+
+        if (CASSANDRA_NAME.equals(adapter_type.toLowerCase()))
+            return getCassandraStore();
+
+        LOG.error("The config could not find a valid value for the field " +
```
Comment (luuse) [inline src/core/StoreSupplier.java]: Same goes for this log message as above.

File: src/core/StoreSupplier.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+    /**
+     * Only for debug at the moment.
+     *
+     * @return Returns a string showing what kind of TsdbStore this supplier
+     * will return when one call {@link StoreSupplier#get()}.
+     */
+    @Override
+    public String toString() {
+        String adapter_type = config.getString("tsd.storage.adapter");
```
Comment (luuse) [inline src/core/StoreSupplier.java]: Can you use the guava to string helpers?

File: test/core/StoreSupplierTest.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+    @Test
+    public void storeSupplierDefaultConfig() {
+        supplier = new StoreSupplier(config);
+        assertTrue(supplier.toString().contains("HBase"));
+    }
+
+    @Test
+    public void storeSupplierChangedConfig() {
+        config.overrideConfig("tsd.storage.adapter", "FOO");
```
Comment (luuse) [inline test/core/StoreSupplierTest.java]: I know that the config can technically change at any time but in the documentation for `#overrideConfig` specifically states that it should only ever be used during initialization. With our strive for making as much as possible immutable in mind i don't necessarily agree with letting the instance provided to `TSDB` to change between calls. At minimum i think we should [memoize](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/Suppliers.html#memoize(com.google.common.base.Supplier)) it.

File: test/core/StoreSupplierTest.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+        assertNull(store);
+    }
+
+    @Test
+    public void testGetInvalid() throws Exception {
+        config.overrideConfig("tsd.storage.adapter", "FooBar4711");
+        supplier = new StoreSupplier(config);
+        try {
+            TsdbStore store = supplier.get();
+            fail("Should have thrown an IllegalArgumentException but did not!");
```
Comment (luuse) [inline test/core/StoreSupplierTest.java]: Why not just just the `expected` variable for the `Test` annotation?

File: src/core/StoreSupplier.java (hunk 1 above)
Comment (jzeeck) [inline src/core/StoreSupplier.java]: Name LOG is used

**Pull request merged**